apps/wolfssh/wolfssh.c

```diff
@@ -991,7 +991,13 @@ static THREAD_RETURN WOLFSSH_THREAD wolfSSH_Client(void* args)
         err_sys("Couldn't set the username.");
 
     build_addr(&clientAddr, config.hostname, config.port);
-    tcp_socket(&sockFd);
+    tcp_socket(&sockFd, 1,
+#ifdef TEST_IPV6
```

Comment:
These checks are really awkward. TEST_IPV6 is meant to force the example tools to use IPV6 for testing. It was a quick-and-dirty way to support IPv6. In general, we shouldn't need checks like this. Information about the IP address type are kept with the IP address. (clientAddr.sin_family)

Reply:
I'm not sure how to make it less awkward.  Since we have the typedef for `SOCKADDR_IN_T` , the check of the member will need this macro wrap. (clientAddr.sin_family vs. clientAddr.sin6_family).